File: backend/express/src/lib/activity-logger.ts
import {db} from "../db/db.ts";
import {activityHistory} from "../db/schema.ts";
import crypto from "crypto";

export async function logActivity(userId: string, description: string) {
  try {
    await db.insert(activityHistory).values({
      id: crypto.randomBytes(16).toString("hex"),
      userId,
      description,
      doneAt: new Date(),
    });
  } catch (error) {
    console.error("Failed to log activity:", error);
  }
}

File: backend/golang/internal/utils/activity_logger.go
package utils

import (
	"log"

	"github.com/maiko26/multi-backend-app/backend/golang/internal/model"
	"github.com/google/uuid"
	"gorm.io/gorm"
)

func LogActivity(db *gorm.DB, email, description string) {
	activity := model.ActivityHistory{
		ID:          uuid.New().String()[:8],
		UserID:      email,
		Description: description,
	}
	if err := db.Create(&activity).Error; err != nil {
		log.Printf("Failed to log activity: %v", err)
	}
}
